File: backend/src/models/Workflow.js
const mongoose = require('mongoose');

const workflowSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  client: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Client',
    required: true
  },
  description: String,
  naturalLanguageInput: {
    type: String,
    required: true
  },
  channels: [{
    type: String,
    enum: ['email', 'sms', 'voice', 'whatsapp', 'webchat'],
    required: true
  }],
  configuration: {
    email: {
      templates: [{
        name: String,
        subject: String,
        content: String,
        triggers: [String]
      }]
    },
    sms: {
      templates: [{
        name: String,
        content: String,
        triggers: [String]
      }]
    },
    voice: {
      ivrFlow: Object,
      scripts: [{
        name: String,
        content: String,
        triggers: [String]
      }]
    },
    webchat: {
      botConfig: Object,
      flowConfig: Object
    }
  },
  zendeskIntegration: {
    enabled: Boolean,
    triggerConditions: [String],
    ticketPriority: String,
    ticketTags: [String]
  },
  status: {
    type: String,
    enum: ['draft', 'active', 'paused', 'archived'],
    default: 'draft'
  },
  metrics: {
    messagesSent: { type: Number, default: 0 },
    callsHandled: { type: Number, default: 0 },
    ticketsCreated: { type: Number, default: 0 },
    estimatedMonthlyCost: Number
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

workflowSchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

module.exports = mongoose.model('Workflow', workflowSchema);

File: backend/models/workflow.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, JSON, ForeignKey, Enum, Text
from sqlalchemy.orm import relationship
import enum
from .base import Base

class WorkflowStatus(enum.Enum):
    ACTIVE = 'active'
    DRAFT = 'draft'
    ARCHIVED = 'archived'

class WorkflowType(enum.Enum):
    SMS = 'sms'
    EMAIL = 'email'
    VOICE = 'voice'

class Workflow(Base):
    __tablename__ = 'workflows'
    __table_args__ = {'extend_existing': True}

    id = Column(String(255), primary_key=True)
    name = Column(String(255), nullable=False)
    description = Column(Text)
    workflow_type = Column(Enum(WorkflowType), nullable=False)
    
    # Configuration Storage
    brand_tone_config = Column(JSON, default={}, nullable=False)  # voice type, greetings, phrases, words to avoid
    ai_training_config = Column(JSON, default={}, nullable=False)  # QA pairs, FAQ docs, chat history
    context_config = Column(JSON, default={}, nullable=False)      # memory window, triggers, knowledge base
    response_config = Column(JSON, default={}, nullable=False)     # templates, message structure
    monitoring_config = Column(JSON, default={}, nullable=False)   # alert thresholds, metrics settings
    
    # Flow Configuration
    nodes = Column(JSON)
    edges = Column(JSON)
    variables = Column(JSON)
    executions = Column(JSON)

    # Metadata
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    business_id = Column(String(255), ForeignKey('businesses.id'))
    status = Column(Enum(WorkflowStatus), default=WorkflowStatus.DRAFT)

    # Relationships
    business = relationship('Business', back_populates='workflows')
    messages = relationship('Message', back_populates='workflow')
    workflow_executions = relationship('WorkflowExecution', back_populates='workflow')
    metrics = relationship('MetricsLog', back_populates='workflow')

    def __repr__(self):
        return f'<Workflow {self.name}>'
